index.js

```diff
@@ -200,173 +215,329 @@ while (args.length) {
-			window.extractCSSOptions = options;
-		}, options);
+			return true;
+		}	
 	}
 
-	if (!height) {
-		var _height = page.evaluate(function () {
-			return document.body.offsetHeight;
-		});
-		page.viewportSize = {
-			width: width,
-			height: _height
-		};
-	}
+	async function injectCssExtractor() {
 
-	if (!fs.isFile(scriptPath)) {
-		fail("Unable to locate script at: " + scriptPath);
-	}
+		if (!html) {
+			html = await page.content();
+		}
 
-	var injection = page.injectJs(scriptPath);
-	if (!injection) {
-		fail("Unable to inject script in page");
-	}
+		if(html.indexOf("stylesheet") === -1) {
+			return false;
+		}
 
-};
+		debug.loadTime = new Date() - debug.loadTime;
+	
+		var options = {};
+	
+		if (matchMQ) {
+			options.matchMQ = true;
+		}
+	
+		if (required) {
+			options.required = required;
+		}
+	
+		if (localStorage) {
+			await page.evaluate(function (data) {
+				var storage = window.localStorage;
+				if (storage) {
+					for (var key in data) {
+						storage.setItem(key, data[key]);
+					}
+				}
+			}, localStorage);
+		}
+	
+		if (Object.keys(options).length) {
+			await page.evaluate(function (options) {
+				window.extractCSSOptions = options;
+			}, options);
+		}
+	
+		if (!height) {
+			var _height = await page.evaluate(function () {
+				return document.body.offsetHeight;
+			});
+			page.viewportSize = {
+				width: width,
+				height: _height
+			};
+		}
+	
+		await page.on("console", async msg => {
 
-if (url) {
+			if (msg.args().length !== 2) {
+				return;
+			}
 
-	debug.loadTime = new Date();
-	page.open(url);
+			if (await msg.args()[0].jsonValue() !== "_extractedcss") {
+				return;
+			}
 
-}
-else {
+			let response = await msg.args()[1].jsonValue();
 
-	if (!fakeUrl) {
-		fail("Missing \"fake-url\" option");
-	}
+			await closePuppeteer();
 
-	html = system.stdin.read();
-	system.stdin.close();
+			await cssExtractorCallback(response);
+		});
 
-	debug.loadTime = new Date();
-	page.setContent(html, fakeUrl);
+		if (!fs.lstatSync(scriptPath).isFile()) {
+			stderr("Unable to locate script at: " + scriptPath);
+			return false;
+		}
+		await page.addScriptTag({path: scriptPath});
 
-}
+		if (browserTimeout) {
+			browserTimeoutHandle = setTimeout(async function () {
+				await closePuppeteer();
+				stderr("Browser timeout");
+			}, browserTimeout);
+		}
 
+		return true;
+	}
 
+	async function cssExtractorCallback(response) {
+		
+		if (!response.css) {
+			stderr("Browser did not return any CSS");
+			return;
+		}
+
+		if ("css" in response) {
+			var result;
+			if (cssOnly) {
+				result = response.css;
+			}
+			else {
+				result = inlineCSS(response.css)
+				if (!result) {
+					return;
+				}
+			}
+			if (outputDebug) {
+				debug.cssLength = response.css.length;
+				debug.time = new Date() - debug.time;
+				debug.processingTime = debug.time - debug.loadTime;
+				result += "\n<!--\n\t" + JSON.stringify(debug) + "\n-->";
+			}
+			if (outputPath) {
+				fs.write(outputPath, result);
```

Comment:
For some reason I get the following error:

```
(node:57908) UnhandledPromiseRejectionWarning: TypeError [ERR_INVALID_ARG_TYPE]: The "fd" argument must be of type number. Received type string
    at Object.write (fs.js:520:3)```

If I switch `fs.write` to `fs.writeFile` there's no error(?).

Reply:
Correct. This seems to be an error. `fs.writeFile`should work.